### PR TITLE
Feature/update gluu Update Gluu client

### PR DESCRIPTION
**Pull request by buiminhducdiag**

### CODE REVIEW

File: src/App/GluuClient.php [hunk 1]
```diff
@@ -4,21 +4,20 @@
...
+    public function __construct($baseUrl, $clientId, $clientSecret)
     {
         $this->gluuHttpClient = new Client([
-            'verify' => false,
-            'base_uri' => $gluuBaseUrl,
-            'auth' => [$gluuClientId, $gluuClientSecret],
-            'http_errors' => true
+            'base_uri' => $baseUrl,
+            'auth' => [$clientId, $clientSecret],
+            'http_errors' => true,
```
Comment (bradleynguyendiag) [inline src/App/GluuClient.php:16]: Remove this

Comment (buiminhducdiag) [reply src/App/GluuClient.php:16]: You meant removing this 'http_errors' => true,' ?

Comment (bradleynguyendiag) [reply src/App/GluuClient.php:16]: Yes. and move this to Provider

File: src/App/GluuClient.php [hunk 2]
```diff
@@ -4,21 +4,20 @@
 
 use GuzzleHttp\Client;
 
-class GluuClient
+class GluuClient implements AuthenticationInterface
 {
     private Client $gluuHttpClient;
 
-    public function __construct($gluuBaseUrl, $gluuClientId, $gluuClientSecret)
+    public function __construct($baseUrl, $clientId, $clientSecret)
```
Comment (bradleynguyendiag) [inline src/App/GluuClient.php:11]:
```suggestion
    public function __construct(private Client $client)
```

File: src/Providers/GluuServiceProvider.php [hunk 3]
```diff
@@ -18,12 +20,12 @@ public function boot()
...
     public function register()
     {
-        // register our controller
-        $this->app->bind(\Gluu\App\GluuClient::class, function ($app) {
-            $gluuBaseUrl = $app['config']->get('gluu.base_url');
-            $gluuClientId = $app['config']->get('gluu.client_id');
-            $gluuClientSecret = $app['config']->get('gluu.client_secret');
-            return new \Gluu\App\GluuClient($gluuBaseUrl, $gluuClientId, $gluuClientSecret);
+        $this->app->bind(AuthenticationInterface::class, function ($app) {
+            return new GluuClient(
```
Comment (bradleynguyendiag) [inline src/Providers/GluuServiceProvider.php:24]:
```suggestion
            return new GluuClient(
            new Client(            'base_uri' => $baseUrl,
            'auth' => [$clientId, $clientSecret])
```

File: src/App/GluuClient.php [hunk 4]
```diff
@@ -4,21 +4,20 @@
...
-            'auth' => [$gluuClientId, $gluuClientSecret],
-            'http_errors' => true
+            'base_uri' => $baseUrl,
+            'auth' => [$clientId, $clientSecret],
+            'http_errors' => true,
         ]);
     }
 
-    public function setGluuHttpClient(Client $client): void
+    public function setClient(Client $client): void
```
Comment (bradleynguyendiag) [inline src/App/GluuClient.php:20]: Remove

File: src/App/GluuClient.php (hunk 2 above)
Comment (buiminhducdiag) [reply src/App/GluuClient.php:11]: Because we are implementing Interface, so we cannot construct with private Client $client

Comment (bradleynguyendiag) [reply src/App/GluuClient.php:11]: @buiminhducdiag Let update AuthenticationInterface, I think better than construct of this one just have Client only.

**Review by bradleynguyendiag — APPROVED** [review]

**Pull request merged**